compiler/rustc_middle/src/traits/mod.rs

```diff
@@ -451,6 +451,9 @@ pub enum ObligationCauseCode<'tcx> {
 
     /// Obligations emitted during the normalization of a weak type alias.
     TypeAlias(InternedObligationCauseCode<'tcx>, Span, DefId),
+
+    /// During borrowck we've found a method call that could have introduced a lifetime requirement.
+    MethodCallConstraint(Ty<'tcx>, Span),
```

Comment:
Please rename this to make it clear that it's a constraint that only ever results from a region constraint category. `MethodCallRegionConstraint` or something would be nice.